File: 프로그래머스/Level2/오픈채팅방.py
# https://programmers.co.kr/learn/courses/30/lessons/42888

def solution(record):
    nick = dict()
    for re in record:
        arr = re.split(' ')
        if arr[0] == 'Enter' or arr[0] == 'Change':
            nick[arr[1]] = arr[2]
    answer = []
    for re in record:
        arr = re.split(' ')
        if arr[0] == 'Enter':
            answer.append(nick[arr[1]] + '님이 들어왔습니다.')
        elif arr[0] == 'Leave':
            answer.append(nick[arr[1]] + '님이 나갔습니다.')
    return answer

File: 프로그래머스/Level2/오픈채팅방.js
// https://programmers.co.kr/learn/courses/30/lessons/42888

const solution = (records) => {
  let answer = [];
  const name = {};
  records.forEach((record) => {
    const temp = record.split(' ');
    if (temp[0] === 'Enter' || temp[0] === 'Change') name[temp[1]] = temp[2];
  });
  records.forEach((record) => {
    const temp = record.split(' ');
    if (temp[0] === 'Enter') answer.push(`${name[temp[1]]}님이 들어왔습니다.`);
    else if (temp[0] === 'Leave') answer.push(`${name[temp[1]]}님이 나갔습니다.`);
  });
  return answer;
};
